pkg/store/bucket.go

```diff
@@ -521,12 +536,32 @@ func (s *BucketStore) TimeRange() (mint, maxt int64) {
 // Info implements the storepb.StoreServer interface.
 func (s *BucketStore) Info(context.Context, *storepb.InfoRequest) (*storepb.InfoResponse, error) {
 	mint, maxt := s.TimeRange()
-	// Store nodes hold global data and thus have no labels.
-	return &storepb.InfoResponse{
+	res := &storepb.InfoResponse{
 		StoreType: component.Store.ToProto(),
 		MinTime:   mint,
 		MaxTime:   maxt,
-	}, nil
+	}
+
+	labelSets := make(map[uint64][]storepb.Label, len(s.blocks))
```

Comment:
Again as I mentioned before - we can build that every sync instead of every Info call. Can we do that? (:

Reply:
Info should be as fast as possible as it works as healthiness/heartbeat check.

Reply:
I will update today. Thanks!